scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +561,87 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    def design_matrix(self, x, kind="dense", check_finite=True):
+        '''
+        Returns a design matrix for BSpline object in "dense" "CSR" formats.
+        
+        Parameters
+        ----------
+        x : 1-D array, shape(n,)
+            Values of x - coordinate of a given set of points (n >= 1).
+        
+        kind : "CSR" or "dense", optional
+            * ``"dense"``: Returns dense matrix.
+            * ``"CSR"``: Returns a tuple ``((data, (row_ind, col_ind)),
+                [n, nt - k - 1])`` suitable for ``csr_matrix`` constructor.
+            Default is ``"dense"``
+            
+        check_finite : bool, optional
+            Whether to check that the input arrays contain only finite numbers.
+            Disabling may give a performance gain, but may result in problems
+            (crashes, non-termination) if the inputs do contain infinities or NaNs.
+            Default is True.    
+        Returns
+        -------
+        2-D array, shape(n, nt - k - 1) or tuple depending on ``kind`` option.
+        '''
```

Comment:
This needs a usage example. It should (a) clarify the dimensions of the result and what exactly is the constructed object (basis elements are in the rows or columns etc), (b) the interaction with `csr_matrix`, and (c) the relation to the OP trick of evaluating a BSpline with a diagonal `c` array.